EIPS/eip-6051.md

```diff
@@ -0,0 +1,464 @@
+
+In addition to `version` string, parameters, involved keys and functions include:
+
+1.  Sender private key  `sk`, to be encapsulated to recipient.
+2.  Ephemeral recipient key pair  `(r, R)`  such that `R = [r]G`. `G` denotes the base point of the elliptic curve, and `[r]G` denotes the scalar multiplication. Optionally, `R` could be signed, and `signerPubKey` and `signature` are then provided for sender to verify if `R`  could be trusted or not.
+3.  Ephemeral sender key pair  `(s, S)`  such that `S = [s]G`.
+4.  Share secret  `ss := [s]R = [r]S` according to ECDH. Note that for secp256k1 this EIP follows RFC5903 and uses compact representation, which means to use *only* the `x` coordinate as the shared secret. For Curve25519 this EIP follows RFC7748.
+5.  `oob`, out of band data, optional. This could be digits or an alpha-numeric string entered by user.
+6.  Let  `derivedKey  :=  HKDF(hash=SHA256, ikm=ss, info=oob,  salt,  length)`. HKDF is defined in RFC5869. The  `length`  should be determined by  `skey`  and  `IV`  requirements such that the symmetric key  `skey  = derivedKey[0:keySize]`, and  `IV  = derivedKey[keySize:length]`.  `keySize`  denotes the key size of underlying symmetric algorithm, for example, 16 (bytes) for AES-128, and 32 (bytes) for Chacha20. See **Security Considerations** for the use of  `salt`.
+7.  `cipher  := authenticated_cncryption(symAlg,  skey,  IV, data=sk)`. The symmetric cipher algorithm `symAlg` and authentication scheme are decided by the version parameter. No additional authentication data `aad` is used.
+
+A much simplified example flow without signature and verification is:
+
+1.  Recipient application generates  `(r, R)`.
+2.  User inputs  `R`  to Sender application, along with a six-digit code “123456” as  `oob`.
+3.  Sender application generates  `(s, S)`, and computes  `cipher`, then returns `S || cipher`
+4.  Recipient application scans to read  `S`  and  `cipher`. User enters “123456” as  `oob`  to Recipient application.
+5.  Recipient application decrypts  `cipher`  to get  `sk`.
+6.  Recipient application derives the address corresponding to  `sk`  so that user can confirm the correctness.
+
+With signature and verification, the signature to `R` by `singerPubKey` is appended to `R`. `signerPubKey` itself could have been already signed by `trustedPubKey`, and that signature is appended to `signerPubKey`. Note that the signature is applied to the byte array data instead of its string representation, which might lead to confusion and interoperability issues (such as hex or base64, lower case v.s. upper case, etc.). See **Requests** and **Test Cases** for further clarification and examples.
+
+### Requests
+#### R1. Request for Recipient to generate ephemeral key pair
+
+```javascript
+request({
+	method: 'eth_generateEphemeralKeyPair',
+	params: [version, signerPubKey],
```

Comment:
Specify the length of version and signerPubKey here or mention it will be specified below.

Reply:
moved "Encoding of data and message" prior to "Requests".